src/reynir_correct/errtokenizer.py

```diff
@@ -2069,21 +2128,23 @@ def is_valid_suggestion(
             assert isinstance(corrected, tuple)
             corrected_display = " ".join(corrected)
             for ix, corrected_word in enumerate(corrected):
-                if ix == 0:
-                    rtok = replace_word(1, token, corrected_word, corrected_display)
-                    at_sentence_start = False
-                else:
-                    # In a multi-word sequence, we only mark the first
-                    # token with a SpellingError
-                    rtok = replace_word(1, token, corrected_word, None)
+                rtok = token
+                if "S001" not in ignore_rules:
```

Comment:
Again, moving the check for ignoring S001 to an outer scope with an early exit is probably better.

Reply:
Done, see comment above